portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,60 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-
+import java.util.ArrayList;
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private ArrayList<String> comments;
 
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+    comments.add("You're so smart");
+    comments.add("Keep up the good work");
+    comments.add(" Nice job! ");
+  }
+    
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+    response.getWriter().println(json);
+  
+}
+ @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "userName", "");
```

Comment:
these are so-called "local variables," they won't persist after this doPost() method finishes. However, you do want them to persist during a servlet instance. You have the "global variable" comments as an ArrayList (in line 28), which you can add the newly input comment to. To make this step starting easier, can you modify the codes so that the comment (just the comment, not the name) you input will show up after you click Hi button?